// @hypercerts-org/contracts/dist/cjs/index.cjs
'use strict';

var TransferManager$6 = {
	address: "0xB3B776a8AAE5BC00c74BB9b3A45Ccc27C6F471F1",
	fullNamespace: "TransferManager",
	args: [
		"0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35"
	],
	encodedArgs: "0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35",
	tx: "0x61ffc042c14063f0c04297350fb9b314559a2d661bc1a38e663639a87b33a5aa"
};
var ProtocolFeeRecipient$4 = {
	address: "0x25C7f07cd5461E95E48aFf54ea7D160Ed8d1E621",
	fullNamespace: "ProtocolFeeRecipient",
	args: [
		"0xe518aED97D9d45174a06bB8EF663B4fB51330725",
		"0x4200000000000000000000000000000000000006"
	],
	encodedArgs: "0xe518aed97d9d45174a06bb8ef663b4fb513307254200000000000000000000000000000000000006",
	tx: "0x74428db26a26f68aabcfd2397ca83d56a5a2f9aa8c6a56f34128e4a88ff88663"
};
var HypercertExchange$5 = {
	address: "0x5DD43eff9FCC6F70564794e997A47722fE315847",
	fullNamespace: "LooksRareProtocol",
	args: [
		"0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35",
		"0x5858053C0A20Ff1cBb00bbb58FAc7A575883F628",
		"0xB3B776a8AAE5BC00c74BB9b3A45Ccc27C6F471F1",
		"0x7b79995e5f793A07Bc00c21412e50Ecae098E7f9"
	],
	encodedArgs: "0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf355858053c0a20ff1cbb00bbb58fac7a575883f628b3b776a8aae5bc00c74bb9b3a45ccc27c6f471f17b79995e5f793a07bc00c21412e50ecae098e7f9",
	tx: "0xbf67d0d028ffbdd9afd63b5ddc457bcb22fd84470642b7120c46bb5cf4e6625f"
};
var RoyaltyFeeRegistry$5 = {
	address: "0x3FAe80819f084DCf27Bf5F9d5002B9246e5Ca6E0",
	fullNamespace: "RoyaltyFeeRegistry",
	args: [
		"1000"
	],
	encodedArgs: "0x00000000000000000000000000000000000000000000000000000000000003e8",
	tx: "0xa6ecc69e62415a01ae2705176676ad7ccd943e6e92e07c91d7384d948e7241ba"
};
var OrderValidator$5 = {
	address: "0x4F931F6954C694636b988f029b21b044aa420586",
	fullNamespace: "OrderValidatorV2A",
	args: [
		"0x5DD43eff9FCC6F70564794e997A47722fE315847"
	],
	encodedArgs: "0x5dd43eff9fcc6f70564794e997a47722fe315847"
};
var CreatorFeeManager$5 = {
	address: "0xe3f64e5fc801a3404129b5afa1aef08f1476d9e5",
	fullNamespace: "CreatorFeeManagerWithRoyalties",
	args: [
		"0x3FAe80819f084DCf27Bf5F9d5002B9246e5Ca6E0"
	],
	encodedArgs: "0x3fae80819f084dcf27bf5f9d5002b9246e5ca6e0",
	tx: "0xa984b0e84602f928371c5b1eaf2374c6f0623932910b114359d118f96233a240"
};
var StrategyHypercertFractionOffer$6 = {
	address: "0xf56ec8eebeed0b3cda7a41a57a3d8e87a17c4f43",
	fullNamespace: "StrategyHypercertFractionOffer",
	args: [
	],
	encodedArgs: "0x",
	tx: "0x433ead67b025cbfa98983b019ca633b56ee159cea7574411434131e6687c3e9f"
};
var deployments_marketplace_base_sepolia = {
	TransferManager: TransferManager$6,
	ProtocolFeeRecipient: ProtocolFeeRecipient$4,
	HypercertExchange: HypercertExchange$5,
	RoyaltyFeeRegistry: RoyaltyFeeRegistry$5,
	OrderValidator: OrderValidator$5,
	CreatorFeeManager: CreatorFeeManager$5,
	StrategyHypercertFractionOffer: StrategyHypercertFractionOffer$6
};

var TransferManager$5 = {
	address: "0x658c1695DCb298E57e6144F6dA3e83DdCF5e2BaB",
	fullNamespace: "TransferManager",
	args: [
		"0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35"
	],
	encodedArgs: "0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35",
	tx: "0x2f991683f500fafe1bbae84dd790d247195f13bf40847dca5db266ad886b607e"
};
var ProtocolFeeRecipient$3 = {
	address: "0x4d99FE0b683874A6FC78A9FA4Fb677AdFc460Bd0",
	fullNamespace: "ProtocolFeeRecipient",
	args: [
		"0xE7C4531ad8828794904D332a12702beC8ff1A498",
		"0x4200000000000000000000000000000000000006"
	],
	encodedArgs: "0xe7c4531ad8828794904d332a12702bec8ff1a4984200000000000000000000000000000000000006",
	tx: "0xbad35b553a07496fb8a9f4b5703957febc5fa6fb3f16849ca42975c5c10cfe7d"
};
var HypercertExchange$4 = {
	address: "0x2F7Ab1844594112E00708e18835ba2e731880Db1",
	fullNamespace: "LooksRareProtocol",
	args: [
		"0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35",
		"0x4d99FE0b683874A6FC78A9FA4Fb677AdFc460Bd0",
		"0x658c1695DCb298E57e6144F6dA3e83DdCF5e2BaB",
		"0x4200000000000000000000000000000000000006"
	],
	encodedArgs: "0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf354d99fe0b683874a6fc78a9fa4fb677adfc460bd0658c1695dcb298e57e6144f6da3e83ddcf5e2bab4200000000000000000000000000000000000006",
	tx: "0xe9f7016b0ca1d9c3c2087ecc42ee8945f120ea5aa9befb0079e023d23d8a15ec"
};
var RoyaltyFeeRegistry$4 = {
	address: "0xd97b92B740EAf655DFf4f93B0D6D38232cfE84C6",
	fullNamespace: "RoyaltyFeeRegistry",
	args: [
		"1000"
	],
	encodedArgs: "0x00000000000000000000000000000000000000000000000000000000000003e8",
	tx: "0x184e33f1ed0dc803a64d1be39758414382393a3797f1a18fe8eb2df1ba249034"
};
var OrderValidator$4 = {
	address: "0x3B51f8c645b6d1894431A11109787c8814D22C32",
	fullNamespace: "OrderValidatorV2A",
	args: [
		"0x2F7Ab1844594112E00708e18835ba2e731880Db1"
	],
	encodedArgs: "0x2f7ab1844594112e00708e18835ba2e731880db1"
};
var CreatorFeeManager$4 = {
	address: "0x2585159c52180f81a7a5b26acea70d398e3a72f5",
	fullNamespace: "CreatorFeeManagerWithRoyalties",
	args: [
		"0xd97b92B740EAf655DFf4f93B0D6D38232cfE84C6"
	],
	encodedArgs: "0xd97b92b740eaf655dff4f93b0d6d38232cfe84c6",
	tx: "0xe72d790b946409a94241eac77bbf6c98be2b7e0bed493842d172ab00b31e2320"
};
var StrategyHypercertFractionOffer$5 = {
	address: "0x9325027afcc9e86285070db2111d48005006026d",
	fullNamespace: "StrategyHypercertFractionOffer",
	args: [
	],
	encodedArgs: "0x",
	tx: "0x96ffab4e6d2d5e068ebeafcaf616586f545346bac25ae7c39a942f0e594ff8c0"
};
var deployments_marketplace_optimism_mainnet = {
	TransferManager: TransferManager$5,
	ProtocolFeeRecipient: ProtocolFeeRecipient$3,
	HypercertExchange: HypercertExchange$4,
	RoyaltyFeeRegistry: RoyaltyFeeRegistry$4,
	OrderValidator: OrderValidator$4,
	CreatorFeeManager: CreatorFeeManager$4,
	StrategyHypercertFractionOffer: StrategyHypercertFractionOffer$5
};

var TransferManager$4 = {
	address: "0x658c1695DCb298E57e6144F6dA3e83DdCF5e2BaB",
	fullNamespace: "TransferManager",
	args: [
		"0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35"
	],
	encodedArgs: "0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35",
	tx: "0x9a4f5d8ab1907a603ebb87cf3e6065a7b205addf10f2ef42c8f8636b2b5ce705"
};
var ProtocolFeeRecipient$2 = {
	address: "0xaDB4b94330dD29F7A7D67156b72077b918E882b4",
	fullNamespace: "ProtocolFeeRecipient",
	args: [
		"0x4f37308832c6eFE5A74737955cBa96257d76De17",
		"0x7b79995e5f793A07Bc00c21412e50Ecae098E7f9"
	],
	encodedArgs: "0x4f37308832c6efe5a74737955cba96257d76de177b79995e5f793a07bc00c21412e50ecae098e7f9",
	tx: "0xfa67db564606ac9a24c0cdffa121ad3ab7e96eefb7383f23d58fbf4db592d428"
};
var HypercertExchange$3 = {
	address: "0xB1991E985197d14669852Be8e53ee95A1f4621c0",
	fullNamespace: "LooksRareProtocol",
	args: [
		"0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35",
		"0xaDB4b94330dD29F7A7D67156b72077b918E882b4",
		"0x658c1695DCb298E57e6144F6dA3e83DdCF5e2BaB",
		"0x7b79995e5f793A07Bc00c21412e50Ecae098E7f9"
	],
	encodedArgs: "0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35adb4b94330dd29f7a7d67156b72077b918e882b4658c1695dcb298e57e6144f6da3e83ddcf5e2bab7b79995e5f793a07bc00c21412e50ecae098e7f9",
	tx: "0x77a1f1284c83a957da90ba7daee7f6085bee34283e221727c9ccc10e8df2027a"
};
var RoyaltyFeeRegistry$3 = {
	address: "0xd97b92B740EAf655DFf4f93B0D6D38232cfE84C6",
	fullNamespace: "RoyaltyFeeRegistry",
	args: [
		"1000"
	],
	encodedArgs: "0x00000000000000000000000000000000000000000000000000000000000003e8",
	tx: "0x21cc81aff04ab92e986dd8e9e1fc5edd9ef7631609800e144def230a1e224129"
};
var OrderValidator$3 = {
	address: "0xB7E30420a456667Aa828722E26fA64729d8DbBce",
	fullNamespace: "OrderValidatorV2A",
	args: [
		"0xB1991E985197d14669852Be8e53ee95A1f4621c0"
	],
	encodedArgs: "0xb1991e985197d14669852be8e53ee95a1f4621c0"
};
var CreatorFeeManager$3 = {
	address: "0xa2ce31fa3204f5ce538723b23f0418e62d54f302",
	fullNamespace: "CreatorFeeManagerWithRoyalties",
	args: [
		"0xd97b92B740EAf655DFf4f93B0D6D38232cfE84C6"
	],
	encodedArgs: "0xd97b92b740eaf655dff4f93b0d6d38232cfe84c6",
	tx: "0x5e9419057ccf80c2b561cca088f2938b0a24293ef434f1965bc56b98502a3797"
};
var StrategyHypercertFractionOffer$4 = {
	address: "0x4a90ef7cc5491eb01319858cca4b1bbce64ff573",
	fullNamespace: "StrategyHypercertFractionOffer",
	args: [
	],
	encodedArgs: "0x",
	tx: "0x20561d2c5d8166cdd6cf955c6041504059df641d7b855521f838cc97d961b616"
};
var deployments_marketplace_sepolia = {
	TransferManager: TransferManager$4,
	ProtocolFeeRecipient: ProtocolFeeRecipient$2,
	HypercertExchange: HypercertExchange$3,
	RoyaltyFeeRegistry: RoyaltyFeeRegistry$3,
	OrderValidator: OrderValidator$3,
	CreatorFeeManager: CreatorFeeManager$3,
	StrategyHypercertFractionOffer: StrategyHypercertFractionOffer$4
};

var deployments_protocol = {
	"10": {
	HypercertMinterUUPS: "0x822F17A9A5EeCFd66dBAFf7946a8071C265D1d07",
	HypercertMinterImplementation: "0xEbA30978164Cc0985091F11532C8f83a5Fa98622"
},
	"314": {
	HypercertMinterUUPS: "0xc756B203cA9e13BAB3a93F1dA756bb19ac3C395b",
	HypercertMinterImplementation: "0xC2d179166bc9dbB00A03686a5b17eCe2224c2704 "
},
	"8453": {
	HypercertMinterUUPS: "0xC2d179166bc9dbB00A03686a5b17eCe2224c2704",
	HypercertMinterImplementation: "0xa16DFb32Eb140a6f3F2AC68f41dAd8c7e83C4941"
},
	"42161": {
	HypercertMinterUUPS: "0x822F17A9A5EeCFd66dBAFf7946a8071C265D1d07",
	HypercertMinterImplementation: "0xc6fbcfe16d5ebfed21ace224c49c94de49046a04"
},
	"42220": {
	HypercertMinterUUPS: "0x16bA53B74c234C870c61EFC04cD418B8f2865959",
	HypercertMinterImplementation: "0x620257e9301a813be7cc5b75aae8dda3f6a926d2"
},
	"84532": {
	HypercertMinterUUPS: "0xC2d179166bc9dbB00A03686a5b17eCe2224c2704",
	HypercertMinterImplementation: "0xa16DFb32Eb140a6f3F2AC68f41dAd8c7e83C4941"
},
	"314159": {
	HypercertMinterUUPS: "0x822F17A9A5EeCFd66dBAFf7946a8071C265D1d07",
	HypercertMinterImplementation: "0xc6FbcFE16D5eBFEd21aCe224C49C94dE49046A04"
},
	"421614": {
	HypercertMinterUUPS: "0x0A00a2f09cd37B24E7429c5238323bfebCfF3Ed9",
	HypercertMinterImplementation: "0x689587461AA3103D3D7975c5e4B352Ab711C14C2"
},
	"11155111": {
	HypercertMinterUUPS: "0xa16DFb32Eb140a6f3F2AC68f41dAd8c7e83C4941",
	HypercertMinterImplementation: "0x0a0df97bddb36eef95fef089a4aeb7aceabf2101"
}
};

var TransferManager$3 = {
	address: "0xf66abbE1c3FE57851010f2Ad5d1A78067fEDE58B",
	fullNamespace: "TransferManager",
	args: [
		"0x5d36971451ae593685cab8815d644f9b4b66ec99"
	],
	encodedArgs: "0x5d36971451ae593685cab8815d644f9b4b66ec99",
	tx: "0x42610c6d7d3766fbbaa25e76d7781865f2983c2a3d88d23540cef384f25ef017"
};
var ProtocolFeeRecipient$1 = {
	address: "0xf9aBC43c85808A720CFC296C5eee4516C7F429F6",
	fullNamespace: "ProtocolFeeRecipient",
	args: [
		"0x5d36971451AE593685Cab8815d644f9B4B66Ec99",
		"0x3031a6D5D9648BA5f50f656Cd4a1672E1167a34A"
	],
	encodedArgs: "0x5d36971451ae593685cab8815d644f9b4b66ec993031a6d5d9648ba5f50f656cd4a1672e1167a34a",
	tx: "0x859ebe02a45909d01850b479b7849ff0d40741db5d50b4a3a5c4c1314dc0434a"
};
var HypercertExchange$2 = {
	address: "0x1d905Bec93E48C64649300688B99D5F7d11ac412",
	fullNamespace: "LooksRareProtocol",
	args: [
		"0x5d36971451ae593685cab8815d644f9b4b66ec99",
		"0xf9aBC43c85808A720CFC296C5eee4516C7F429F6",
		"0xf66abbE1c3FE57851010f2Ad5d1A78067fEDE58B",
		"0x3031a6D5D9648BA5f50f656Cd4a1672E1167a34A"
	],
	encodedArgs: "0x5d36971451ae593685cab8815d644f9b4b66ec99f9abc43c85808a720cfc296c5eee4516c7f429f6f66abbe1c3fe57851010f2ad5d1a78067fede58b3031a6d5d9648ba5f50f656cd4a1672e1167a34a",
	tx: "0xeec15808a49fec00f53906ef9ad71ab40521fe8fb30103293957a0ef332d8f30"
};
var RoyaltyFeeRegistry$2 = {
	address: "0xfCb6A37b57497E4418058BbdEfc680e2e3fe2a7f",
	fullNamespace: "RoyaltyFeeRegistry",
	args: [
		"1000"
	],
	encodedArgs: "0x00000000000000000000000000000000000000000000000000000000000003e8",
	tx: "0x48dbc49140fbdfba587ff538522b5fd904b33b89c52eebdd2a859d23e7d8564d"
};
var OrderValidator$2 = {
	address: "0x976427894fAE68821289f6D71388802273CAc33e",
	fullNamespace: "OrderValidatorV2A",
	args: [
		"0x1d905Bec93E48C64649300688B99D5F7d11ac412"
	],
	encodedArgs: "0x1d905bec93e48c64649300688b99d5f7d11ac412"
};
var CreatorFeeManager$2 = {
	address: "0xa3d5438ba3d65677d8531340da201d93e06eb5b7",
	fullNamespace: "CreatorFeeManagerWithRoyalties",
	args: [
		"0xfCb6A37b57497E4418058BbdEfc680e2e3fe2a7f"
	],
	encodedArgs: "0xfcb6a37b57497e4418058bbdefc680e2e3fe2a7f",
	tx: "0xe18a640375b419caa7d71bcee5ab5fb0681fd83142ef99c3aa4112944ce6e690"
};
var StrategyHypercertFractionOffer$3 = {
	address: "0x58fc599e0afd4e526a5ff69933d7a00721d28bb1",
	fullNamespace: "StrategyHypercertFractionOffer",
	args: [
	],
	encodedArgs: "0x",
	tx: "0xcafff6f303d32fcd7d34a6c235b573edc27a31f4c2f2e332d492b2414cae106b"
};
var deployments_marketplace_arb_sepolia = {
	TransferManager: TransferManager$3,
	ProtocolFeeRecipient: ProtocolFeeRecipient$1,
	HypercertExchange: HypercertExchange$2,
	RoyaltyFeeRegistry: RoyaltyFeeRegistry$2,
	OrderValidator: OrderValidator$2,
	CreatorFeeManager: CreatorFeeManager$2,
	StrategyHypercertFractionOffer: StrategyHypercertFractionOffer$3
};

var TransferManager$2 = {
	address: "0x658c1695DCb298E57e6144F6dA3e83DdCF5e2BaB",
	fullNamespace: "TransferManager",
	args: [
		"0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35"
	],
	encodedArgs: "0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35",
	tx: "0x70b306371278c66189814b465e151006ad6a6e0c9d6b910c0234ac2ec8da1706"
};
var ProtocolFeeRecipient = {
	address: "0xE332d4b0a30a522099292594999967267480a081",
	fullNamespace: "ProtocolFeeRecipient",
	args: [
		"0xE7C4531ad8828794904D332a12702beC8ff1A498",
		"0x82aF49447D8a07e3bd95BD0d56f35241523fBab1"
	],
	encodedArgs: "0xe7c4531ad8828794904d332a12702bec8ff1a49882af49447d8a07e3bd95bd0d56f35241523fbab1",
	tx: "0x6d9ff2423d04879b3445bb95c2abdfe760090391440e1e2ea92e26c267d5c696"
};
var HypercertExchange$1 = {
	address: "0xcE8fa09562f07c23B9C21b5d0A29a293F8a8BC83",
	fullNamespace: "LooksRareProtocol",
	args: [
		"0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35",
		"0xE332d4b0a30a522099292594999967267480a081",
		"0x658c1695DCb298E57e6144F6dA3e83DdCF5e2BaB",
		"0x82aF49447D8a07e3bd95BD0d56f35241523fBab1"
	],
	encodedArgs: "0xdc6d6f9ab5fcc398b92b017e8482749ae5afbf35e332d4b0a30a522099292594999967267480a081658c1695dcb298e57e6144f6da3e83ddcf5e2bab82af49447d8a07e3bd95bd0d56f35241523fbab1",
	tx: "0x1eff2ebd3d6ad2a97f92c4caa648fb34fb25ca60730625d3491cd2b1171fcc23"
};
var RoyaltyFeeRegistry$1 = {
	address: "0xd97b92B740EAf655DFf4f93B0D6D38232cfE84C6",
	fullNamespace: "RoyaltyFeeRegistry",
	args: [
		"1000"
	],
	encodedArgs: "0x00000000000000000000000000000000000000000000000000000000000003e8",
	tx: "0x8352d52116ff97551dca27801b0fe1995e9b66ad45062021c13da4d7855f9658"
};
var OrderValidator$1 = {
	address: "0x74b05A917BBa5050ea216877c81801d6049b3f16",
	fullNamespace: "OrderValidatorV2A",
	args: [
		"0xcE8fa09562f07c23B9C21b5d0A29a293F8a8BC83"
	],
	encodedArgs: "0xce8fa09562f07c23b9c21b5d0a29a293f8a8bc83"
};
var CreatorFeeManager$1 = {
	address: "0x2d25df25ab95b74a18eb2d719a98b47f5e5f5534",
	fullNamespace: "CreatorFeeManagerWithRoyalties",
	args: [
		"0xd97b92B740EAf655DFf4f93B0D6D38232cfE84C6"
	],
	encodedArgs: "0xd97b92b740eaf655dff4f93b0d6d38232cfe84c6",
	tx: "0x2339dcf7dda93e4fbfc763a70dfe932b793552ee805c584d10acdf6127ba18f2"
};
var StrategyHypercertFractionOffer$2 = {
	address: "0xecab24cade0261fc6513ca13bb3d10f760af3da8",
	fullNamespace: "StrategyHypercertFractionOffer",
	args: [
	],
	encodedArgs: "0x",
	tx: "0x79012a15888ec5c77e7c68cbc85614645f13603e18a1f951746fa5857c0f0a23"
};
var deployments_marketplace_arb_one = {
	TransferManager: TransferManager$2,
	ProtocolFeeRecipient: ProtocolFeeRecipient,
	HypercertExchange: HypercertExchange$1,
	RoyaltyFeeRegistry: RoyaltyFeeRegistry$1,
	OrderValidator: OrderValidator$1,
	CreatorFeeManager: CreatorFeeManager$1,
	StrategyHypercertFractionOffer: StrategyHypercertFractionOffer$2
};

var TransferManager$1 = {
	address: "0xd64827b10f019EB2A45E407C785231FE969D7805",
	fullNamespace: "TransferManager",
	args: [
		"0xb92fd98b5085d686944ff997772d3d8e37e28573"
	],
	encodedArgs: "0xb92fd98b5085d686944ff997772d3d8e37e28573",
	tx: "0x41a3925dc2f4163d28d1d4ea3fed496dbba73e0ccbff26ed268e5d2a96728814"
};
var HypercertExchange = {
	address: "0xfB29FDb0677b1acB4b148F953DE5ee6ec5cBf717",
	fullNamespace: "LooksRareProtocol",
	args: [
		"0xb92fd98b5085d686944ff997772d3d8e37e28573",
		"0xE7C4531ad8828794904D332a12702beC8ff1A498",
		"0xd64827b10f019EB2A45E407C785231FE969D7805",
		"0x0000000000000000000000000000000000000000"
	],
	encodedArgs: "0xb92fd98b5085d686944ff997772d3d8e37e28573e7c4531ad8828794904d332a12702bec8ff1a498d64827b10f019eb2a45e407c785231fe969d78050000000000000000000000000000000000000000",
	tx: "0x4b3d34dc62d4634df09bf6cd69938c781371f6b06ab9baf025872b03e2733d74"
};
var RoyaltyFeeRegistry = {
	address: "0xee550d3E9F741f187D5957C75937dbeE94278Db2",
	fullNamespace: "RoyaltyFeeRegistry",
	args: [
		"1000"
	],
	encodedArgs: "0x00000000000000000000000000000000000000000000000000000000000003e8",
	tx: "0xb08d9232ba8cd2845854058569817d8289f73a08550bebcb16fd18e17cecffd8"
};
var OrderValidator = {
	address: "0x1c4654558674e5b0Aa9C33F3A74F2EfA83d0C35e",
	fullNamespace: "OrderValidatorV2A",
	args: [
		"0xfB29FDb0677b1acB4b148F953DE5ee6ec5cBf717"
	],
	encodedArgs: "0xfb29fdb0677b1acb4b148f953de5ee6ec5cbf717"
};
var CreatorFeeManager = {
	address: "0x5bd2524c95ad3033e86d84a29d1cc52207f08df7",
	fullNamespace: "CreatorFeeManagerWithRoyalties",
	args: [
		"0xee550d3E9F741f187D5957C75937dbeE94278Db2"
	],
	encodedArgs: "0xee550d3e9f741f187d5957c75937dbee94278db2",
	tx: "0x16c6de547199655e056785aca8419191c3fa48b226b4de4c06b636a168831a32"
};
var StrategyHypercertFractionOffer$1 = {
	address: "0xc80b087d36acab5a4ececba203b589bb3c5606ee",
	fullNamespace: "StrategyHypercertFractionOffer",
	args: [
	],
	encodedArgs: "0x",
	tx: "0xfc515150190f135ba7b24554c94eda602fdeb092e55a0807af30b45418c5938c"
};
var deployments_marketplace_celo_mainnet = {
	TransferManager: TransferManager$1,
	HypercertExchange: HypercertExchange,
	RoyaltyFeeRegistry: RoyaltyFeeRegistry,
	OrderValidator: OrderValidator,
	CreatorFeeManager: CreatorFeeManager,
	StrategyHypercertFractionOffer: StrategyHypercertFractionOffer$1
};

const deployments_marketplace = {
    "10": {
        TransferManager: deployments_marketplace_optimism_mainnet.TransferManager.address,
        HypercertExchange: deployments_marketplace_optimism_mainnet.HypercertExchange.address,
        OrderValidatorV2A: deployments_marketplace_optimism_mainnet.OrderValidator.address,
        RoyaltyFeeRegistry: deployments_marketplace_optimism_mainnet.RoyaltyFeeRegistry.address,
        StrategyHypercertFractionOffer: deployments_marketplace_optimism_mainnet.StrategyHypercertFractionOffer.address
    },
    "84532": {
        TransferManager: deployments_marketplace_base_sepolia.TransferManager.address,
        HypercertExchange: deployments_marketplace_base_sepolia.HypercertExchange.address,
        OrderValidatorV2A: deployments_marketplace_base_sepolia.OrderValidator.address,
        RoyaltyFeeRegistry: deployments_marketplace_base_sepolia.RoyaltyFeeRegistry.address,
        StrategyHypercertFractionOffer: deployments_marketplace_base_sepolia.StrategyHypercertFractionOffer.address
    },
    "11155111": {
        TransferManager: deployments_marketplace_sepolia.TransferManager.address,
        HypercertExchange: deployments_marketplace_sepolia.HypercertExchange.address,
        OrderValidatorV2A: deployments_marketplace_sepolia.OrderValidator.address,
        RoyaltyFeeRegistry: deployments_marketplace_sepolia.RoyaltyFeeRegistry.address,
        StrategyHypercertFractionOffer: deployments_marketplace_sepolia.StrategyHypercertFractionOffer.address
    },
    "421614": {
        TransferManager: deployments_marketplace_arb_sepolia.TransferManager.address,
        HypercertExchange: deployments_marketplace_arb_sepolia.HypercertExchange.address,
        OrderValidatorV2A: deployments_marketplace_arb_sepolia.OrderValidator.address,
        RoyaltyFeeRegistry: deployments_marketplace_arb_sepolia.RoyaltyFeeRegistry.address,
        StrategyHypercertFractionOffer: deployments_marketplace_arb_sepolia.StrategyHypercertFractionOffer.address
    },
    "42161": {
        TransferManager: deployments_marketplace_arb_one.TransferManager.address,
        HypercertExchange: deployments_marketplace_arb_one.HypercertExchange.address,
        OrderValidatorV2A: deployments_marketplace_arb_one.OrderValidator.address,
        RoyaltyFeeRegistry: deployments_marketplace_arb_one.RoyaltyFeeRegistry.address,
        StrategyHypercertFractionOffer: deployments_marketplace_arb_one.StrategyHypercertFractionOffer.address
    },
    "42220": {
        TransferManager: deployments_marketplace_celo_mainnet.TransferManager.address,
        HypercertExchange: deployments_marketplace_celo_mainnet.HypercertExchange.address,
        OrderValidatorV2A: deployments_marketplace_celo_mainnet.OrderValidator.address,
        RoyaltyFeeRegistry: deployments_marketplace_celo_mainnet.RoyaltyFeeRegistry.address,
        StrategyHypercertFractionOffer: deployments_marketplace_celo_mainnet.StrategyHypercertFractionOffer.address
    }
};
var DEPLOYMENTS = {
    marketplace: deployments_marketplace,
    protocol: deployments_protocol
};

var HypercertMinter = [
	{
		inputs: [
		],
		stateMutability: "nonpayable",
		type: "constructor"
	},
	{
		inputs: [
		],
		name: "AlreadyClaimed",
		type: "error"
	},
	{
		inputs: [
		],
		name: "ArraySize",
		type: "error"
	},
	{
		inputs: [
		],
		name: "DoesNotExist",
		type: "error"
	},
	{
		inputs: [
		],
		name: "DuplicateEntry",
		type: "error"
	},
	{
		inputs: [
		],
		name: "Invalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotAllowed",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotApprovedOrOwner",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransfersNotAllowed",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TypeMismatch",
		type: "error"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "previousAdmin",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address",
				name: "newAdmin",
				type: "address"
			}
		],
		name: "AdminChanged",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "tokenID",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "bytes32",
				name: "root",
				type: "bytes32"
			}
		],
		name: "AllowlistCreated",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: true,
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				indexed: true,
				internalType: "address",
				name: "operator",
				type: "address"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "approved",
				type: "bool"
			}
		],
		name: "ApprovalForAll",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256[]",
				name: "claimIDs",
				type: "uint256[]"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "fromTokenIDs",
				type: "uint256[]"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "toTokenIDs",
				type: "uint256[]"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "values",
				type: "uint256[]"
			}
		],
		name: "BatchValueTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: true,
				internalType: "address",
				name: "beacon",
				type: "address"
			}
		],
		name: "BeaconUpgraded",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: true,
				internalType: "uint256",
				name: "claimID",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "string",
				name: "uri",
				type: "string"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "totalUnits",
				type: "uint256"
			}
		],
		name: "ClaimStored",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint8",
				name: "version",
				type: "uint8"
			}
		],
		name: "Initialized",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "tokenID",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "bytes32",
				name: "leaf",
				type: "bytes32"
			}
		],
		name: "LeafClaimed",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: true,
				internalType: "address",
				name: "previousOwner",
				type: "address"
			},
			{
				indexed: true,
				internalType: "address",
				name: "newOwner",
				type: "address"
			}
		],
		name: "OwnershipTransferred",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "account",
				type: "address"
			}
		],
		name: "Paused",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: true,
				internalType: "address",
				name: "operator",
				type: "address"
			},
			{
				indexed: true,
				internalType: "address",
				name: "from",
				type: "address"
			},
			{
				indexed: true,
				internalType: "address",
				name: "to",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "ids",
				type: "uint256[]"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "values",
				type: "uint256[]"
			}
		],
		name: "TransferBatch",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: true,
				internalType: "address",
				name: "operator",
				type: "address"
			},
			{
				indexed: true,
				internalType: "address",
				name: "from",
				type: "address"
			},
			{
				indexed: true,
				internalType: "address",
				name: "to",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "id",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "value",
				type: "uint256"
			}
		],
		name: "TransferSingle",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "string",
				name: "value",
				type: "string"
			},
			{
				indexed: true,
				internalType: "uint256",
				name: "id",
				type: "uint256"
			}
		],
		name: "URI",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "account",
				type: "address"
			}
		],
		name: "Unpaused",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: true,
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		name: "Upgraded",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "claimID",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "fromTokenID",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "toTokenID",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "value",
				type: "uint256"
			}
		],
		name: "ValueTransfer",
		type: "event"
	},
	{
		inputs: [
		],
		name: "__SemiFungible1155_init",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "id",
				type: "uint256"
			}
		],
		name: "balanceOf",
		outputs: [
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address[]",
				name: "accounts",
				type: "address[]"
			},
			{
				internalType: "uint256[]",
				name: "ids",
				type: "uint256[]"
			}
		],
		name: "balanceOfBatch",
		outputs: [
			{
				internalType: "uint256[]",
				name: "",
				type: "uint256[]"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "_account",
				type: "address"
			},
			{
				internalType: "uint256[]",
				name: "_tokenIDs",
				type: "uint256[]"
			}
		],
		name: "batchBurnFraction",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "bytes32[][]",
				name: "proofs",
				type: "bytes32[][]"
			},
			{
				internalType: "uint256[]",
				name: "claimIDs",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "units",
				type: "uint256[]"
			}
		],
		name: "batchMintClaimsFromAllowlists",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "id",
				type: "uint256"
			},
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		name: "burn",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "uint256[]",
				name: "ids",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "",
				type: "uint256[]"
			}
		],
		name: "burnBatch",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "_account",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "_tokenID",
				type: "uint256"
			}
		],
		name: "burnFraction",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "units",
				type: "uint256"
			},
			{
				internalType: "bytes32",
				name: "merkleRoot",
				type: "bytes32"
			},
			{
				internalType: "string",
				name: "_uri",
				type: "string"
			},
			{
				internalType: "enum IHypercertToken.TransferRestrictions",
				name: "restrictions",
				type: "uint8"
			}
		],
		name: "createAllowlist",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "claimID",
				type: "uint256"
			}
		],
		name: "getMinted",
		outputs: [
			{
				internalType: "uint256",
				name: "mintedUnits",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			},
			{
				internalType: "bytes32",
				name: "",
				type: "bytes32"
			}
		],
		name: "hasBeenClaimed",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "initialize",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "bytes32[]",
				name: "proof",
				type: "bytes32[]"
			},
			{
				internalType: "uint256",
				name: "claimID",
				type: "uint256"
			},
			{
				internalType: "bytes32",
				name: "leaf",
				type: "bytes32"
			}
		],
		name: "isAllowedToClaim",
		outputs: [
			{
				internalType: "bool",
				name: "isAllowed",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "address",
				name: "operator",
				type: "address"
			}
		],
		name: "isApprovedForAll",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "_account",
				type: "address"
			},
			{
				internalType: "uint256[]",
				name: "_fractionIDs",
				type: "uint256[]"
			}
		],
		name: "mergeFractions",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "units",
				type: "uint256"
			},
			{
				internalType: "string",
				name: "_uri",
				type: "string"
			},
			{
				internalType: "enum IHypercertToken.TransferRestrictions",
				name: "restrictions",
				type: "uint8"
			}
		],
		name: "mintClaim",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "bytes32[]",
				name: "proof",
				type: "bytes32[]"
			},
			{
				internalType: "uint256",
				name: "claimID",
				type: "uint256"
			},
			{
				internalType: "uint256",
				name: "units",
				type: "uint256"
			}
		],
		name: "mintClaimFromAllowlist",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "units",
				type: "uint256"
			},
			{
				internalType: "uint256[]",
				name: "fractions",
				type: "uint256[]"
			},
			{
				internalType: "string",
				name: "_uri",
				type: "string"
			},
			{
				internalType: "enum IHypercertToken.TransferRestrictions",
				name: "restrictions",
				type: "uint8"
			}
		],
		name: "mintClaimWithFractions",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "name",
		outputs: [
			{
				internalType: "string",
				name: "",
				type: "string"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "owner",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "tokenID",
				type: "uint256"
			}
		],
		name: "ownerOf",
		outputs: [
			{
				internalType: "address",
				name: "_owner",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "pause",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "paused",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "proxiableUUID",
		outputs: [
			{
				internalType: "bytes32",
				name: "",
				type: "bytes32"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "tokenID",
				type: "uint256"
			}
		],
		name: "readTransferRestriction",
		outputs: [
			{
				internalType: "string",
				name: "",
				type: "string"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "renounceOwnership",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "from",
				type: "address"
			},
			{
				internalType: "address",
				name: "to",
				type: "address"
			},
			{
				internalType: "uint256[]",
				name: "ids",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			},
			{
				internalType: "bytes",
				name: "data",
				type: "bytes"
			}
		],
		name: "safeBatchTransferFrom",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "from",
				type: "address"
			},
			{
				internalType: "address",
				name: "to",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "id",
				type: "uint256"
			},
			{
				internalType: "uint256",
				name: "amount",
				type: "uint256"
			},
			{
				internalType: "bytes",
				name: "data",
				type: "bytes"
			}
		],
		name: "safeTransferFrom",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "operator",
				type: "address"
			},
			{
				internalType: "bool",
				name: "approved",
				type: "bool"
			}
		],
		name: "setApprovalForAll",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "_account",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "_tokenID",
				type: "uint256"
			},
			{
				internalType: "uint256[]",
				name: "_newFractions",
				type: "uint256[]"
			}
		],
		name: "splitFraction",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "bytes4",
				name: "interfaceId",
				type: "bytes4"
			}
		],
		name: "supportsInterface",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newOwner",
				type: "address"
			}
		],
		name: "transferOwnership",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "account",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "tokenID",
				type: "uint256"
			}
		],
		name: "unitsOf",
		outputs: [
			{
				internalType: "uint256",
				name: "units",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "tokenID",
				type: "uint256"
			}
		],
		name: "unitsOf",
		outputs: [
			{
				internalType: "uint256",
				name: "units",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "unpause",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newImplementation",
				type: "address"
			}
		],
		name: "upgradeTo",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newImplementation",
				type: "address"
			},
			{
				internalType: "bytes",
				name: "data",
				type: "bytes"
			}
		],
		name: "upgradeToAndCall",
		outputs: [
		],
		stateMutability: "payable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "tokenID",
				type: "uint256"
			}
		],
		name: "uri",
		outputs: [
			{
				internalType: "string",
				name: "_uri",
				type: "string"
			}
		],
		stateMutability: "view",
		type: "function"
	}
];

var LooksRareProtocol = [
	{
		inputs: [
			{
				internalType: "address",
				name: "_owner",
				type: "address"
			},
			{
				internalType: "address",
				name: "_protocolFeeRecipient",
				type: "address"
			},
			{
				internalType: "address",
				name: "_transferManager",
				type: "address"
			},
			{
				internalType: "address",
				name: "_weth",
				type: "address"
			}
		],
		stateMutability: "nonpayable",
		type: "constructor"
	},
	{
		inputs: [
		],
		name: "CallerInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "ChainIdInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "CollectionTypeInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "CreatorFeeBpTooHigh",
		type: "error"
	},
	{
		inputs: [
		],
		name: "CurrencyInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "ERC20TransferFromFail",
		type: "error"
	},
	{
		inputs: [
		],
		name: "LengthsInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "MerkleProofInvalid",
		type: "error"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "length",
				type: "uint256"
			}
		],
		name: "MerkleProofTooLarge",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NewGasLimitETHTransferTooLow",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NewProtocolFeeRecipientCannotBeNullAddress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NoOngoingTransferInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NoSelectorForStrategy",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NoncesInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotAContract",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotOwner",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotV2Strategy",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NullSignerAddress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OutsideOfTimeRange",
		type: "error"
	},
	{
		inputs: [
		],
		name: "QuoteTypeInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "ReentrancyFail",
		type: "error"
	},
	{
		inputs: [
		],
		name: "RenouncementNotInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "SameDomainSeparator",
		type: "error"
	},
	{
		inputs: [
		],
		name: "SignatureEOAInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "SignatureERC1271Invalid",
		type: "error"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "length",
				type: "uint256"
			}
		],
		name: "SignatureLengthInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "SignatureParameterSInvalid",
		type: "error"
	},
	{
		inputs: [
			{
				internalType: "uint8",
				name: "v",
				type: "uint8"
			}
		],
		name: "SignatureParameterVInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "StrategyHasNoSelector",
		type: "error"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			}
		],
		name: "StrategyNotAvailable",
		type: "error"
	},
	{
		inputs: [
		],
		name: "StrategyNotUsed",
		type: "error"
	},
	{
		inputs: [
		],
		name: "StrategyProtocolFeeTooHigh",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransferAlreadyInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransferNotInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "UnitAmountInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "WrongPotentialOwner",
		type: "error"
	},
	{
		anonymous: false,
		inputs: [
		],
		name: "CancelOwnershipTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "currency",
				type: "address"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "isAllowed",
				type: "bool"
			}
		],
		name: "CurrencyStatusUpdated",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
		],
		name: "InitiateOwnershipRenouncement",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "previousOwner",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address",
				name: "potentialOwner",
				type: "address"
			}
		],
		name: "InitiateOwnershipTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "user",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "bidNonce",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "askNonce",
				type: "uint256"
			}
		],
		name: "NewBidAskNonces",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "creatorFeeManager",
				type: "address"
			}
		],
		name: "NewCreatorFeeManager",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
		],
		name: "NewDomainSeparator",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "gasLimitETHTransfer",
				type: "uint256"
			}
		],
		name: "NewGasLimitETHTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "maxCreatorFeeBp",
				type: "uint256"
			}
		],
		name: "NewMaxCreatorFeeBp",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "newOwner",
				type: "address"
			}
		],
		name: "NewOwner",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "protocolFeeRecipient",
				type: "address"
			}
		],
		name: "NewProtocolFeeRecipient",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "maxProtocolFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "bytes4",
				name: "selector",
				type: "bytes4"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "isMakerBid",
				type: "bool"
			},
			{
				indexed: false,
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		name: "NewStrategy",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "user",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "orderNonces",
				type: "uint256[]"
			}
		],
		name: "OrderNoncesCancelled",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "isActive",
				type: "bool"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			}
		],
		name: "StrategyUpdated",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "user",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "subsetNonces",
				type: "uint256[]"
			}
		],
		name: "SubsetNoncesCancelled",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				components: [
					{
						internalType: "bytes32",
						name: "orderHash",
						type: "bytes32"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "bool",
						name: "isNonceInvalidated",
						type: "bool"
					}
				],
				indexed: false,
				internalType: "struct ILooksRareProtocol.NonceInvalidationParameters",
				name: "nonceInvalidationParameters",
				type: "tuple"
			},
			{
				indexed: false,
				internalType: "address",
				name: "askUser",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address",
				name: "bidUser",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "address",
				name: "currency",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address",
				name: "collection",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			},
			{
				indexed: false,
				internalType: "address[2]",
				name: "feeRecipients",
				type: "address[2]"
			},
			{
				indexed: false,
				internalType: "uint256[3]",
				name: "feeAmounts",
				type: "uint256[3]"
			}
		],
		name: "TakerAsk",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				components: [
					{
						internalType: "bytes32",
						name: "orderHash",
						type: "bytes32"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "bool",
						name: "isNonceInvalidated",
						type: "bool"
					}
				],
				indexed: false,
				internalType: "struct ILooksRareProtocol.NonceInvalidationParameters",
				name: "nonceInvalidationParameters",
				type: "tuple"
			},
			{
				indexed: false,
				internalType: "address",
				name: "bidUser",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address",
				name: "bidRecipient",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "address",
				name: "currency",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address",
				name: "collection",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			},
			{
				indexed: false,
				internalType: "address[2]",
				name: "feeRecipients",
				type: "address[2]"
			},
			{
				indexed: false,
				internalType: "uint256[3]",
				name: "feeAmounts",
				type: "uint256[3]"
			}
		],
		name: "TakerBid",
		type: "event"
	},
	{
		inputs: [
		],
		name: "MAGIC_VALUE_ORDER_NONCE_EXECUTED",
		outputs: [
			{
				internalType: "bytes32",
				name: "",
				type: "bytes32"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "WETH",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "collection",
				type: "address"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			}
		],
		name: "_getUnitsHeldByHypercertFractions",
		outputs: [
			{
				internalType: "uint256",
				name: "unitsHeldByItems",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "maxProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "bytes4",
				name: "selector",
				type: "bytes4"
			},
			{
				internalType: "bool",
				name: "isMakerBid",
				type: "bool"
			},
			{
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		name: "addStrategy",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256[]",
				name: "orderNonces",
				type: "uint256[]"
			}
		],
		name: "cancelOrderNonces",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "cancelOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256[]",
				name: "subsetNonces",
				type: "uint256[]"
			}
		],
		name: "cancelSubsetNonces",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "chainId",
		outputs: [
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "confirmOwnershipRenouncement",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "confirmOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "creatorFeeManager",
		outputs: [
			{
				internalType: "contract ICreatorFeeManager",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "domainSeparator",
		outputs: [
			{
				internalType: "bytes32",
				name: "",
				type: "bytes32"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker[]",
				name: "takerBids",
				type: "tuple[]"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker[]",
				name: "makerAsks",
				type: "tuple[]"
			},
			{
				internalType: "bytes[]",
				name: "makerSignatures",
				type: "bytes[]"
			},
			{
				components: [
					{
						internalType: "bytes32",
						name: "root",
						type: "bytes32"
					},
					{
						components: [
							{
								internalType: "bytes32",
								name: "value",
								type: "bytes32"
							},
							{
								internalType: "enum OrderStructs.MerkleTreeNodePosition",
								name: "position",
								type: "uint8"
							}
						],
						internalType: "struct OrderStructs.MerkleTreeNode[]",
						name: "proof",
						type: "tuple[]"
					}
				],
				internalType: "struct OrderStructs.MerkleTree[]",
				name: "merkleTrees",
				type: "tuple[]"
			},
			{
				internalType: "bool",
				name: "isAtomic",
				type: "bool"
			}
		],
		name: "executeMultipleTakerBids",
		outputs: [
		],
		stateMutability: "payable",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker",
				name: "takerAsk",
				type: "tuple"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerBid",
				type: "tuple"
			},
			{
				internalType: "bytes",
				name: "makerSignature",
				type: "bytes"
			},
			{
				components: [
					{
						internalType: "bytes32",
						name: "root",
						type: "bytes32"
					},
					{
						components: [
							{
								internalType: "bytes32",
								name: "value",
								type: "bytes32"
							},
							{
								internalType: "enum OrderStructs.MerkleTreeNodePosition",
								name: "position",
								type: "uint8"
							}
						],
						internalType: "struct OrderStructs.MerkleTreeNode[]",
						name: "proof",
						type: "tuple[]"
					}
				],
				internalType: "struct OrderStructs.MerkleTree",
				name: "merkleTree",
				type: "tuple"
			}
		],
		name: "executeTakerAsk",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker",
				name: "takerBid",
				type: "tuple"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerAsk",
				type: "tuple"
			},
			{
				internalType: "bytes",
				name: "makerSignature",
				type: "bytes"
			},
			{
				components: [
					{
						internalType: "bytes32",
						name: "root",
						type: "bytes32"
					},
					{
						components: [
							{
								internalType: "bytes32",
								name: "value",
								type: "bytes32"
							},
							{
								internalType: "enum OrderStructs.MerkleTreeNodePosition",
								name: "position",
								type: "uint8"
							}
						],
						internalType: "struct OrderStructs.MerkleTreeNode[]",
						name: "proof",
						type: "tuple[]"
					}
				],
				internalType: "struct OrderStructs.MerkleTree",
				name: "merkleTree",
				type: "tuple"
			}
		],
		name: "executeTakerBid",
		outputs: [
		],
		stateMutability: "payable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "bytes32",
				name: "root",
				type: "bytes32"
			},
			{
				internalType: "uint256",
				name: "proofLength",
				type: "uint256"
			}
		],
		name: "hashBatchOrder",
		outputs: [
			{
				internalType: "bytes32",
				name: "batchOrderHash",
				type: "bytes32"
			}
		],
		stateMutability: "pure",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "bool",
				name: "bid",
				type: "bool"
			},
			{
				internalType: "bool",
				name: "ask",
				type: "bool"
			}
		],
		name: "incrementBidAskNonces",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "initiateOwnershipRenouncement",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newPotentialOwner",
				type: "address"
			}
		],
		name: "initiateOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		name: "isCurrencyAllowed",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "maxCreatorFeeBp",
		outputs: [
			{
				internalType: "uint16",
				name: "",
				type: "uint16"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "owner",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "ownershipStatus",
		outputs: [
			{
				internalType: "enum IOwnableTwoSteps.Status",
				name: "",
				type: "uint8"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "potentialOwner",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "protocolFeeRecipient",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker",
				name: "takerBid",
				type: "tuple"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerAsk",
				type: "tuple"
			},
			{
				internalType: "address",
				name: "sender",
				type: "address"
			},
			{
				internalType: "bytes32",
				name: "orderHash",
				type: "bytes32"
			}
		],
		name: "restrictedExecuteTakerBid",
		outputs: [
			{
				internalType: "uint256",
				name: "protocolFeeAmount",
				type: "uint256"
			}
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		name: "strategyInfo",
		outputs: [
			{
				internalType: "bool",
				name: "isActive",
				type: "bool"
			},
			{
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "maxProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "bytes4",
				name: "selector",
				type: "bytes4"
			},
			{
				internalType: "bool",
				name: "isMakerBid",
				type: "bool"
			},
			{
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "transferManager",
		outputs: [
			{
				internalType: "contract TransferManager",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newCreatorFeeManager",
				type: "address"
			}
		],
		name: "updateCreatorFeeManager",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "currency",
				type: "address"
			},
			{
				internalType: "bool",
				name: "isAllowed",
				type: "bool"
			}
		],
		name: "updateCurrencyStatus",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "updateDomainSeparator",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "newGasLimitETHTransfer",
				type: "uint256"
			}
		],
		name: "updateETHGasLimitForTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint16",
				name: "newMaxCreatorFeeBp",
				type: "uint16"
			}
		],
		name: "updateMaxCreatorFeeBp",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newProtocolFeeRecipient",
				type: "address"
			}
		],
		name: "updateProtocolFeeRecipient",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				internalType: "bool",
				name: "isActive",
				type: "bool"
			},
			{
				internalType: "uint16",
				name: "newStandardProtocolFee",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "newMinTotalFee",
				type: "uint16"
			}
		],
		name: "updateStrategy",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		name: "userBidAskNonces",
		outputs: [
			{
				internalType: "uint256",
				name: "bidNonce",
				type: "uint256"
			},
			{
				internalType: "uint256",
				name: "askNonce",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		name: "userOrderNonce",
		outputs: [
			{
				internalType: "bytes32",
				name: "",
				type: "bytes32"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		name: "userSubsetNonce",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	}
];

var OrderValidatorV2A = [
	{
		inputs: [
			{
				internalType: "address",
				name: "_looksRareProtocol",
				type: "address"
			}
		],
		stateMutability: "nonpayable",
		type: "constructor"
	},
	{
		inputs: [
		],
		name: "CRITERIA_GROUPS",
		outputs: [
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "ERC1155_INTERFACE_ID",
		outputs: [
			{
				internalType: "bytes4",
				name: "",
				type: "bytes4"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "ERC721_INTERFACE_ID_1",
		outputs: [
			{
				internalType: "bytes4",
				name: "",
				type: "bytes4"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "ERC721_INTERFACE_ID_2",
		outputs: [
			{
				internalType: "bytes4",
				name: "",
				type: "bytes4"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "HYPERCERT_INTERFACE_ID",
		outputs: [
			{
				internalType: "bytes4",
				name: "",
				type: "bytes4"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "MAGIC_VALUE_ORDER_NONCE_EXECUTED",
		outputs: [
			{
				internalType: "bytes32",
				name: "",
				type: "bytes32"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerOrder",
				type: "tuple"
			},
			{
				internalType: "bytes",
				name: "signature",
				type: "bytes"
			},
			{
				components: [
					{
						internalType: "bytes32",
						name: "root",
						type: "bytes32"
					},
					{
						components: [
							{
								internalType: "bytes32",
								name: "value",
								type: "bytes32"
							},
							{
								internalType: "enum OrderStructs.MerkleTreeNodePosition",
								name: "position",
								type: "uint8"
							}
						],
						internalType: "struct OrderStructs.MerkleTreeNode[]",
						name: "proof",
						type: "tuple[]"
					}
				],
				internalType: "struct OrderStructs.MerkleTree",
				name: "merkleTree",
				type: "tuple"
			}
		],
		name: "checkMakerOrderValidity",
		outputs: [
			{
				internalType: "uint256[9]",
				name: "validationCodes",
				type: "uint256[9]"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker[]",
				name: "makerOrders",
				type: "tuple[]"
			},
			{
				internalType: "bytes[]",
				name: "signatures",
				type: "bytes[]"
			},
			{
				components: [
					{
						internalType: "bytes32",
						name: "root",
						type: "bytes32"
					},
					{
						components: [
							{
								internalType: "bytes32",
								name: "value",
								type: "bytes32"
							},
							{
								internalType: "enum OrderStructs.MerkleTreeNodePosition",
								name: "position",
								type: "uint8"
							}
						],
						internalType: "struct OrderStructs.MerkleTreeNode[]",
						name: "proof",
						type: "tuple[]"
					}
				],
				internalType: "struct OrderStructs.MerkleTree[]",
				name: "merkleTrees",
				type: "tuple[]"
			}
		],
		name: "checkMultipleMakerOrderValidities",
		outputs: [
			{
				internalType: "uint256[9][]",
				name: "validationCodes",
				type: "uint256[9][]"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "creatorFeeManager",
		outputs: [
			{
				internalType: "contract ICreatorFeeManager",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "deriveProtocolParameters",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "domainSeparator",
		outputs: [
			{
				internalType: "bytes32",
				name: "",
				type: "bytes32"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "looksRareProtocol",
		outputs: [
			{
				internalType: "contract LooksRareProtocol",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "maxCreatorFeeBp",
		outputs: [
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "transferManager",
		outputs: [
			{
				internalType: "contract TransferManager",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	}
];

var StrategyManager = [
	{
		inputs: [
			{
				internalType: "address",
				name: "_owner",
				type: "address"
			}
		],
		stateMutability: "nonpayable",
		type: "constructor"
	},
	{
		inputs: [
		],
		name: "NoOngoingTransferInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotOwner",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotV2Strategy",
		type: "error"
	},
	{
		inputs: [
		],
		name: "RenouncementNotInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "StrategyHasNoSelector",
		type: "error"
	},
	{
		inputs: [
		],
		name: "StrategyNotUsed",
		type: "error"
	},
	{
		inputs: [
		],
		name: "StrategyProtocolFeeTooHigh",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransferAlreadyInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransferNotInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "WrongPotentialOwner",
		type: "error"
	},
	{
		anonymous: false,
		inputs: [
		],
		name: "CancelOwnershipTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "currency",
				type: "address"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "isAllowed",
				type: "bool"
			}
		],
		name: "CurrencyStatusUpdated",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
		],
		name: "InitiateOwnershipRenouncement",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "previousOwner",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address",
				name: "potentialOwner",
				type: "address"
			}
		],
		name: "InitiateOwnershipTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "newOwner",
				type: "address"
			}
		],
		name: "NewOwner",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "maxProtocolFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "bytes4",
				name: "selector",
				type: "bytes4"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "isMakerBid",
				type: "bool"
			},
			{
				indexed: false,
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		name: "NewStrategy",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "isActive",
				type: "bool"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			}
		],
		name: "StrategyUpdated",
		type: "event"
	},
	{
		inputs: [
			{
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "maxProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "bytes4",
				name: "selector",
				type: "bytes4"
			},
			{
				internalType: "bool",
				name: "isMakerBid",
				type: "bool"
			},
			{
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		name: "addStrategy",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "cancelOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "confirmOwnershipRenouncement",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "confirmOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "initiateOwnershipRenouncement",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newPotentialOwner",
				type: "address"
			}
		],
		name: "initiateOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		name: "isCurrencyAllowed",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "owner",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "ownershipStatus",
		outputs: [
			{
				internalType: "enum IOwnableTwoSteps.Status",
				name: "",
				type: "uint8"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "potentialOwner",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		name: "strategyInfo",
		outputs: [
			{
				internalType: "bool",
				name: "isActive",
				type: "bool"
			},
			{
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "maxProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "bytes4",
				name: "selector",
				type: "bytes4"
			},
			{
				internalType: "bool",
				name: "isMakerBid",
				type: "bool"
			},
			{
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "currency",
				type: "address"
			},
			{
				internalType: "bool",
				name: "isAllowed",
				type: "bool"
			}
		],
		name: "updateCurrencyStatus",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				internalType: "bool",
				name: "isActive",
				type: "bool"
			},
			{
				internalType: "uint16",
				name: "newStandardProtocolFee",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "newMinTotalFee",
				type: "uint16"
			}
		],
		name: "updateStrategy",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	}
];

var TransferManager = [
	{
		inputs: [
			{
				internalType: "address",
				name: "_owner",
				type: "address"
			}
		],
		stateMutability: "nonpayable",
		type: "constructor"
	},
	{
		inputs: [
		],
		name: "AmountInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "ERC1155SafeBatchTransferFromFail",
		type: "error"
	},
	{
		inputs: [
		],
		name: "ERC1155SafeTransferFromFail",
		type: "error"
	},
	{
		inputs: [
		],
		name: "ERC721TransferFromFail",
		type: "error"
	},
	{
		inputs: [
		],
		name: "HypercertSplitFractionError",
		type: "error"
	},
	{
		inputs: [
		],
		name: "LengthsInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NoOngoingTransferInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotAContract",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotAContract",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotOwner",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OperatorAlreadyAllowed",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OperatorAlreadyApprovedByUser",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OperatorNotAllowed",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OperatorNotApprovedByUser",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OrderInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "RenouncementNotInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransferAlreadyInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransferCallerInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransferNotInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "WrongPotentialOwner",
		type: "error"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "user",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address[]",
				name: "operators",
				type: "address[]"
			}
		],
		name: "ApprovalsGranted",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "user",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address[]",
				name: "operators",
				type: "address[]"
			}
		],
		name: "ApprovalsRemoved",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
		],
		name: "CancelOwnershipTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
		],
		name: "InitiateOwnershipRenouncement",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "previousOwner",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address",
				name: "potentialOwner",
				type: "address"
			}
		],
		name: "InitiateOwnershipTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "newOwner",
				type: "address"
			}
		],
		name: "NewOwner",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "operator",
				type: "address"
			}
		],
		name: "OperatorAllowed",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "operator",
				type: "address"
			}
		],
		name: "OperatorRemoved",
		type: "event"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "operator",
				type: "address"
			}
		],
		name: "allowOperator",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "cancelOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "confirmOwnershipRenouncement",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "confirmOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address[]",
				name: "operators",
				type: "address[]"
			}
		],
		name: "grantApprovals",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			},
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		name: "hasUserApprovedOperator",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "initiateOwnershipRenouncement",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newPotentialOwner",
				type: "address"
			}
		],
		name: "initiateOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		name: "isOperatorAllowed",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "owner",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "ownershipStatus",
		outputs: [
			{
				internalType: "enum IOwnableTwoSteps.Status",
				name: "",
				type: "uint8"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "potentialOwner",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "operator",
				type: "address"
			}
		],
		name: "removeOperator",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address[]",
				name: "operators",
				type: "address[]"
			}
		],
		name: "revokeApprovals",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "collection",
				type: "address"
			},
			{
				internalType: "address",
				name: "from",
				type: "address"
			},
			{
				internalType: "address",
				name: "to",
				type: "address"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			}
		],
		name: "splitItemsHypercert",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					}
				],
				internalType: "struct ITransferManager.BatchTransferItem[]",
				name: "items",
				type: "tuple[]"
			},
			{
				internalType: "address",
				name: "from",
				type: "address"
			},
			{
				internalType: "address",
				name: "to",
				type: "address"
			}
		],
		name: "transferBatchItemsAcrossCollections",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "collection",
				type: "address"
			},
			{
				internalType: "address",
				name: "from",
				type: "address"
			},
			{
				internalType: "address",
				name: "to",
				type: "address"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			}
		],
		name: "transferItemsERC1155",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "collection",
				type: "address"
			},
			{
				internalType: "address",
				name: "from",
				type: "address"
			},
			{
				internalType: "address",
				name: "to",
				type: "address"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			}
		],
		name: "transferItemsERC721",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "collection",
				type: "address"
			},
			{
				internalType: "address",
				name: "from",
				type: "address"
			},
			{
				internalType: "address",
				name: "to",
				type: "address"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			}
		],
		name: "transferItemsHypercert",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	}
];

var StrategyHypercertFractionOffer = [
	{
		inputs: [
		],
		name: "AmountInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "LengthsInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "MerkleProofInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OrderInvalid",
		type: "error"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker",
				name: "takerBid",
				type: "tuple"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerAsk",
				type: "tuple"
			}
		],
		name: "executeHypercertFractionStrategyWithTakerBid",
		outputs: [
			{
				internalType: "uint256",
				name: "price",
				type: "uint256"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			},
			{
				internalType: "bool",
				name: "isNonceInvalidated",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker",
				name: "takerBid",
				type: "tuple"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerAsk",
				type: "tuple"
			}
		],
		name: "executeHypercertFractionStrategyWithTakerBidWithAllowlist",
		outputs: [
			{
				internalType: "uint256",
				name: "price",
				type: "uint256"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			},
			{
				internalType: "bool",
				name: "isNonceInvalidated",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "isLooksRareV2Strategy",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "pure",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerAsk",
				type: "tuple"
			},
			{
				internalType: "bytes4",
				name: "functionSelector",
				type: "bytes4"
			}
		],
		name: "isMakerOrderValid",
		outputs: [
			{
				internalType: "bool",
				name: "isValid",
				type: "bool"
			},
			{
				internalType: "bytes4",
				name: "errorSelector",
				type: "bytes4"
			}
		],
		stateMutability: "view",
		type: "function"
	}
];

var CreatorFeeManagerWithRoyalties = [
	{
		inputs: [
			{
				internalType: "address",
				name: "_royaltyFeeRegistry",
				type: "address"
			}
		],
		stateMutability: "nonpayable",
		type: "constructor"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "collection",
				type: "address"
			}
		],
		name: "BundleEIP2981NotAllowed",
		type: "error"
	},
	{
		inputs: [
		],
		name: "royaltyFeeRegistry",
		outputs: [
			{
				internalType: "contract IRoyaltyFeeRegistry",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "collection",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "price",
				type: "uint256"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			}
		],
		name: "viewCreatorFeeInfo",
		outputs: [
			{
				internalType: "address",
				name: "creator",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "creatorFeeAmount",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	}
];

var StrategyHypercertCollectionOffer = [
	{
		inputs: [
		],
		name: "CollectionTypeInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "MerkleProofInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OrderInvalid",
		type: "error"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker",
				name: "takerAsk",
				type: "tuple"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerBid",
				type: "tuple"
			}
		],
		name: "executeHypercertCollectionStrategyWithTakerAsk",
		outputs: [
			{
				internalType: "uint256",
				name: "price",
				type: "uint256"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			},
			{
				internalType: "bool",
				name: "isNonceInvalidated",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker",
				name: "takerAsk",
				type: "tuple"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerBid",
				type: "tuple"
			}
		],
		name: "executeHypercertCollectionStrategyWithTakerAskWithAllowlist",
		outputs: [
			{
				internalType: "uint256",
				name: "price",
				type: "uint256"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			},
			{
				internalType: "bool",
				name: "isNonceInvalidated",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker",
				name: "takerAsk",
				type: "tuple"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerBid",
				type: "tuple"
			}
		],
		name: "executeHypercertCollectionStrategyWithTakerAskWithProof",
		outputs: [
			{
				internalType: "uint256",
				name: "price",
				type: "uint256"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			},
			{
				internalType: "bool",
				name: "isNonceInvalidated",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "isLooksRareV2Strategy",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "pure",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerBid",
				type: "tuple"
			},
			{
				internalType: "bytes4",
				name: "functionSelector",
				type: "bytes4"
			}
		],
		name: "isMakerOrderValid",
		outputs: [
			{
				internalType: "bool",
				name: "isValid",
				type: "bool"
			},
			{
				internalType: "bytes4",
				name: "errorSelector",
				type: "bytes4"
			}
		],
		stateMutability: "pure",
		type: "function"
	}
];

var StrategyHypercertDutchAuction = [
	{
		inputs: [
		],
		name: "BidTooLow",
		type: "error"
	},
	{
		inputs: [
		],
		name: "CollectionTypeInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OrderInvalid",
		type: "error"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "address",
						name: "recipient",
						type: "address"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Taker",
				name: "takerBid",
				type: "tuple"
			},
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerAsk",
				type: "tuple"
			}
		],
		name: "executeStrategyWithTakerBid",
		outputs: [
			{
				internalType: "uint256",
				name: "price",
				type: "uint256"
			},
			{
				internalType: "uint256[]",
				name: "itemIds",
				type: "uint256[]"
			},
			{
				internalType: "uint256[]",
				name: "amounts",
				type: "uint256[]"
			},
			{
				internalType: "bool",
				name: "isNonceInvalidated",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "isLooksRareV2Strategy",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "pure",
		type: "function"
	},
	{
		inputs: [
			{
				components: [
					{
						internalType: "enum QuoteType",
						name: "quoteType",
						type: "uint8"
					},
					{
						internalType: "uint256",
						name: "globalNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "subsetNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "orderNonce",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "strategyId",
						type: "uint256"
					},
					{
						internalType: "enum CollectionType",
						name: "collectionType",
						type: "uint8"
					},
					{
						internalType: "address",
						name: "collection",
						type: "address"
					},
					{
						internalType: "address",
						name: "currency",
						type: "address"
					},
					{
						internalType: "address",
						name: "signer",
						type: "address"
					},
					{
						internalType: "uint256",
						name: "startTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "endTime",
						type: "uint256"
					},
					{
						internalType: "uint256",
						name: "price",
						type: "uint256"
					},
					{
						internalType: "uint256[]",
						name: "itemIds",
						type: "uint256[]"
					},
					{
						internalType: "uint256[]",
						name: "amounts",
						type: "uint256[]"
					},
					{
						internalType: "bytes",
						name: "additionalParameters",
						type: "bytes"
					}
				],
				internalType: "struct OrderStructs.Maker",
				name: "makerAsk",
				type: "tuple"
			},
			{
				internalType: "bytes4",
				name: "functionSelector",
				type: "bytes4"
			}
		],
		name: "isMakerOrderValid",
		outputs: [
			{
				internalType: "bool",
				name: "isValid",
				type: "bool"
			},
			{
				internalType: "bytes4",
				name: "errorSelector",
				type: "bytes4"
			}
		],
		stateMutability: "view",
		type: "function"
	}
];

var ExecutionManager = [
	{
		inputs: [
			{
				internalType: "address",
				name: "_owner",
				type: "address"
			},
			{
				internalType: "address",
				name: "_protocolFeeRecipient",
				type: "address"
			}
		],
		stateMutability: "nonpayable",
		type: "constructor"
	},
	{
		inputs: [
		],
		name: "CreatorFeeBpTooHigh",
		type: "error"
	},
	{
		inputs: [
		],
		name: "LengthsInvalid",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NewProtocolFeeRecipientCannotBeNullAddress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NoOngoingTransferInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NoSelectorForStrategy",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotOwner",
		type: "error"
	},
	{
		inputs: [
		],
		name: "NotV2Strategy",
		type: "error"
	},
	{
		inputs: [
		],
		name: "OutsideOfTimeRange",
		type: "error"
	},
	{
		inputs: [
		],
		name: "RenouncementNotInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "StrategyHasNoSelector",
		type: "error"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			}
		],
		name: "StrategyNotAvailable",
		type: "error"
	},
	{
		inputs: [
		],
		name: "StrategyNotUsed",
		type: "error"
	},
	{
		inputs: [
		],
		name: "StrategyProtocolFeeTooHigh",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransferAlreadyInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "TransferNotInProgress",
		type: "error"
	},
	{
		inputs: [
		],
		name: "WrongPotentialOwner",
		type: "error"
	},
	{
		anonymous: false,
		inputs: [
		],
		name: "CancelOwnershipTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "currency",
				type: "address"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "isAllowed",
				type: "bool"
			}
		],
		name: "CurrencyStatusUpdated",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
		],
		name: "InitiateOwnershipRenouncement",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "previousOwner",
				type: "address"
			},
			{
				indexed: false,
				internalType: "address",
				name: "potentialOwner",
				type: "address"
			}
		],
		name: "InitiateOwnershipTransfer",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "user",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "bidNonce",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "uint256",
				name: "askNonce",
				type: "uint256"
			}
		],
		name: "NewBidAskNonces",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "creatorFeeManager",
				type: "address"
			}
		],
		name: "NewCreatorFeeManager",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "maxCreatorFeeBp",
				type: "uint256"
			}
		],
		name: "NewMaxCreatorFeeBp",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "newOwner",
				type: "address"
			}
		],
		name: "NewOwner",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "protocolFeeRecipient",
				type: "address"
			}
		],
		name: "NewProtocolFeeRecipient",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "maxProtocolFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "bytes4",
				name: "selector",
				type: "bytes4"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "isMakerBid",
				type: "bool"
			},
			{
				indexed: false,
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		name: "NewStrategy",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "user",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "orderNonces",
				type: "uint256[]"
			}
		],
		name: "OrderNoncesCancelled",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				indexed: false,
				internalType: "bool",
				name: "isActive",
				type: "bool"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				indexed: false,
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			}
		],
		name: "StrategyUpdated",
		type: "event"
	},
	{
		anonymous: false,
		inputs: [
			{
				indexed: false,
				internalType: "address",
				name: "user",
				type: "address"
			},
			{
				indexed: false,
				internalType: "uint256[]",
				name: "subsetNonces",
				type: "uint256[]"
			}
		],
		name: "SubsetNoncesCancelled",
		type: "event"
	},
	{
		inputs: [
		],
		name: "MAGIC_VALUE_ORDER_NONCE_EXECUTED",
		outputs: [
			{
				internalType: "bytes32",
				name: "",
				type: "bytes32"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "maxProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "bytes4",
				name: "selector",
				type: "bytes4"
			},
			{
				internalType: "bool",
				name: "isMakerBid",
				type: "bool"
			},
			{
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		name: "addStrategy",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256[]",
				name: "orderNonces",
				type: "uint256[]"
			}
		],
		name: "cancelOrderNonces",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "cancelOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256[]",
				name: "subsetNonces",
				type: "uint256[]"
			}
		],
		name: "cancelSubsetNonces",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "confirmOwnershipRenouncement",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "confirmOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "creatorFeeManager",
		outputs: [
			{
				internalType: "contract ICreatorFeeManager",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "bool",
				name: "bid",
				type: "bool"
			},
			{
				internalType: "bool",
				name: "ask",
				type: "bool"
			}
		],
		name: "incrementBidAskNonces",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
		],
		name: "initiateOwnershipRenouncement",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newPotentialOwner",
				type: "address"
			}
		],
		name: "initiateOwnershipTransfer",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		name: "isCurrencyAllowed",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "maxCreatorFeeBp",
		outputs: [
			{
				internalType: "uint16",
				name: "",
				type: "uint16"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "owner",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "ownershipStatus",
		outputs: [
			{
				internalType: "enum IOwnableTwoSteps.Status",
				name: "",
				type: "uint8"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "potentialOwner",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
		],
		name: "protocolFeeRecipient",
		outputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		name: "strategyInfo",
		outputs: [
			{
				internalType: "bool",
				name: "isActive",
				type: "bool"
			},
			{
				internalType: "uint16",
				name: "standardProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "minTotalFeeBp",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "maxProtocolFeeBp",
				type: "uint16"
			},
			{
				internalType: "bytes4",
				name: "selector",
				type: "bytes4"
			},
			{
				internalType: "bool",
				name: "isMakerBid",
				type: "bool"
			},
			{
				internalType: "address",
				name: "implementation",
				type: "address"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newCreatorFeeManager",
				type: "address"
			}
		],
		name: "updateCreatorFeeManager",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "currency",
				type: "address"
			},
			{
				internalType: "bool",
				name: "isAllowed",
				type: "bool"
			}
		],
		name: "updateCurrencyStatus",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint16",
				name: "newMaxCreatorFeeBp",
				type: "uint16"
			}
		],
		name: "updateMaxCreatorFeeBp",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "newProtocolFeeRecipient",
				type: "address"
			}
		],
		name: "updateProtocolFeeRecipient",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "uint256",
				name: "strategyId",
				type: "uint256"
			},
			{
				internalType: "bool",
				name: "isActive",
				type: "bool"
			},
			{
				internalType: "uint16",
				name: "newStandardProtocolFee",
				type: "uint16"
			},
			{
				internalType: "uint16",
				name: "newMinTotalFee",
				type: "uint16"
			}
		],
		name: "updateStrategy",
		outputs: [
		],
		stateMutability: "nonpayable",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			}
		],
		name: "userBidAskNonces",
		outputs: [
			{
				internalType: "uint256",
				name: "bidNonce",
				type: "uint256"
			},
			{
				internalType: "uint256",
				name: "askNonce",
				type: "uint256"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		name: "userOrderNonce",
		outputs: [
			{
				internalType: "bytes32",
				name: "",
				type: "bytes32"
			}
		],
		stateMutability: "view",
		type: "function"
	},
	{
		inputs: [
			{
				internalType: "address",
				name: "",
				type: "address"
			},
			{
				internalType: "uint256",
				name: "",
				type: "uint256"
			}
		],
		name: "userSubsetNonce",
		outputs: [
			{
				internalType: "bool",
				name: "",
				type: "bool"
			}
		],
		stateMutability: "view",
		type: "function"
	}
];

// Deployments
const deployments = {
    10: {
        ...DEPLOYMENTS.protocol["10"],
        ...DEPLOYMENTS.marketplace["10"]
    },
    42220: {
        ...DEPLOYMENTS.protocol[42220],
        ...DEPLOYMENTS.marketplace["42220"]
    },
    11155111: {
        ...DEPLOYMENTS.protocol["11155111"],
        ...DEPLOYMENTS.marketplace["11155111"]
    },
    84532: {
        ...DEPLOYMENTS.protocol["84532"],
        ...DEPLOYMENTS.marketplace["84532"]
    },
    8453: {
        ...DEPLOYMENTS.protocol["8453"]
    },
    42161: {
        ...DEPLOYMENTS.protocol["42161"],
        ...DEPLOYMENTS.marketplace["42161"]
    },
    421614: {
        ...DEPLOYMENTS.protocol["421614"],
        ...DEPLOYMENTS.marketplace["421614"]
    },
    314159: {
        ...DEPLOYMENTS.protocol["314159"]
    },
    314: {
        ...DEPLOYMENTS.protocol["314"]
    }
};
const asDeployedChain = (chainId)=>{
    if (chainId in deployments) return chainId;
    throw new Error(`Chain ${chainId} not deployed`);
};

exports.CreatorFeeManagerWithRoyaltiesAbi = CreatorFeeManagerWithRoyalties;
exports.ExecutionManagerAbi = ExecutionManager;
exports.HypercertExchangeAbi = LooksRareProtocol;
exports.HypercertMinterAbi = HypercertMinter;
exports.OrderValidatorV2AAbi = OrderValidatorV2A;
exports.StrategyHypercertCollectionOfferAbi = StrategyHypercertCollectionOffer;
exports.StrategyHypercertDutchAuctionAbi = StrategyHypercertDutchAuction;
exports.StrategyHypercertFractionOfferAbi = StrategyHypercertFractionOffer;
exports.StrategyManagerAbi = StrategyManager;
exports.TransferManagerAbi = TransferManager;
exports.asDeployedChain = asDeployedChain;
exports.deployments = deployments;
